lib/fizzy/execute.cpp

```diff
@@ -701,7 +701,13 @@ std::unique_ptr<Instance> instantiate(Module module,
 
     auto [table, table_limits] = allocate_table(module.tablesec, imported_tables);
 
-    auto [memory, memory_limits] = allocate_memory(module.memorysec, imported_memories);
+    auto [memory, memory_limits] =
+        allocate_memory(module.memorysec, imported_memories, memory_pages_limit);
+    if (memory_limits.max.has_value())
```

Comment:
Would add something like:
```suggestion
    // If imported memory is used, we need to adjust the memory limit.
    // Note: allocate_memory ensures this limit is always below memory_pages_limit.
    if (memory_limits.max.has_value())
```

Reply:
I added the comment, but it applies to non-imported memory, too.

Reply:
Ah right, it is the memory section setting.